common/include/pcl/common/impl/io.hpp

```diff
@@ -59,7 +59,16 @@ template <typename PointT> int
 pcl::getFieldIndex (const std::string &field_name, 
                     std::vector<pcl::PCLPointField> &fields)
 {
-  getFields<PointT> (fields);
+  fields = getFields<PointT> ();
+  const auto& ref = fields;
```

Comment:
Could you please explain this?

Reply:
Essentially, `fields` get populated with the fields everytime this function is called. So I created a const overload which as implied by the contract can't populate the `field`, merely find in the field. The non-const overload still has the old behavior, and the const version works much better with return value `getFields` when used with const.